client/blocks/product-selector/index.jsx

```diff
@@ -183,28 +184,33 @@ export class ProductSelector extends Component {
 		} );
 	}
 
-	handleCheckoutForProduct = productObject => {
-		const { selectedSiteSlug } = this.props;
-
-		return () => {
-			page( '/checkout/' + selectedSiteSlug + '/' + productObject.product_slug );
-		};
-	};
-
 	handleProductOptionSelect( stateKey, productSlug ) {
 		this.setState( {
 			[ stateKey ]: productSlug,
 		} );
 	}
 
 	renderCheckoutButton( product ) {
-		const { intervalType, storeProducts, translate } = this.props;
+		const {
+			currentPlanSlug,
+			intervalType,
+			selectedSiteSlug,
+			storeProducts,
+			translate,
+		} = this.props;
 		const selectedProductSlug = this.state[ this.getStateKey( product.id, intervalType ) ];
 		const productObject = storeProducts[ selectedProductSlug ];
 
 		return (
 			<ProductCardAction
-				onClick={ this.handleCheckoutForProduct( productObject ) }
+				onClick={ () => {
```

Comment:
I don't think we should be defining this on every re-render. Let's make it a component method. Should we have just reused the `handleCheckoutForProduct` method?

Reply:
I agree. I wanted to get away from some issues that I had when passing a `product` object to the `handleCheckoutForProduct` method but it is no longer the case.

I'm going to revive the `handleCheckoutForProduct` method in my next commit.